### ISSUE TITLE
Get very large rmse...

### ISSUE THREAD

**Issue by wangli-coder** [Microsoft/LightGBM#1928]
I wrote the following code to train a lightGBM model, I got a very large rmse value, and my model can make a correct prediction.
Can someone answer my doubts? By the way, forgive me for my bad English.Thanks a lot.

code:
    x_train, x_eval, y_train, y_eval = train_test_split(
        train_data, train_y, test_size=0.3, random_state=0)
    lgb_train = lgb.Dataset(x_train, y_train)
    lgb_eval = lgb.Dataset(x_eval, y_eval)
    params = {'boosting_type' : 'gbdt',
              'metric' : 'rmse',
              'objective' : 'regression_l2',
              'learning_rate' : 0.1,
              'num_leaves' : 31,
              'max_depth' : 6,
              'bagging_fraction' : 0.8,
              'bagging_freq' : 10,
              'feature_fraction' : 0.8,
              'lambda_l2' : 0.3,
              'lambda_l1' : 0.3}
    model = lgb.train(train_set=lgb_train, valid_sets=lgb_eval, params=params,
                      num_boost_round=2000, early_stopping_rounds=100)

output:
[418]	valid_0's rmse: 28123.1
[419]	valid_0's rmse: 28120.5
[420]	valid_0's rmse: 28135.1
[421]	valid_0's rmse: 28115.7
[422]	valid_0's rmse: 28103.7
[423]	valid_0's rmse: 28100.9
[424]	valid_0's rmse: 28082.4
[425]	valid_0's rmse: 28072.7
[426]	valid_0's rmse: 28079.7
[427]	valid_0's rmse: 28079.8
[428]	valid_0's rmse: 28060.8
[429]	valid_0's rmse: 28061.4
[430]	valid_0's rmse: 28037.4
[431]	valid_0's rmse: 28038.4
[432]	valid_0's rmse: 28046.3
[433]	valid_0's rmse: 28040.9
[434]	valid_0's rmse: 28047.9
[435]	valid_0's rmse: 28043.2
[436]	valid_0's rmse: 28057.5
[437]	valid_0's rmse: 28059.2
[438]	valid_0's rmse: 28051
[439]	valid_0's rmse: 28068.6
[440]	valid_0's rmse: 28074.9
[441]	valid_0's rmse: 28067.3
[442]	valid_0's rmse: 28071.8
[443]	valid_0's rmse: 28075.3
[444]	valid_0's rmse: 28080.6
[445]	valid_0's rmse: 28088.1
[446]	valid_0's rmse: 28091.1
[447]	valid_0's rmse: 28078.6
[448]	valid_0's rmse: 28071.7
Early stopping, best iteration is:
[348]	valid_0's rmse: 27989.6

**Comment by guolinke** [comment]
what is the average value of labels in your validation data?

**Comment by wangli-coder** [comment]
> what is the average value of labels in your validation data?

The average value of labels in my validation data is 181702.78

**Comment by guolinke** [comment]
@wangli-coder 
When all labels are large, you can try to set `reg_sqrt=false`.

**Comment by wangli-coder** [comment]
@mayer79 Thank you for your suggestion. When I converted the label to a logarithm, rmse became normal. I can also use rmsle to calculate the error.^_^
